### PR TITLE
Remove old violation_count method

### PR DESCRIPTION
**Pull request by gylaz**

### CODE REVIEW

**Comment by thorncp** [comment]
👍 ✂️

**Comment by teoljungberg** [comment]
Do we really want this? Don't we want to update the build status with the violation count if there are any?

**Comment by thorncp** [comment]
I assumed "old violation_count method" meant it wasn't used anymore, but the Circle failure says otherwise. Do we have a replacement for this method?

**Comment by gylaz** [comment]
There is a `violations_count` column on `builds` that replaces this.

**Comment by teoljungberg** [comment]
@gylaz In that case, we'd need to update the reporter to use that method.

File: app/services/build_report.rb [hunk 1]
```diff
@@ -37,14 +37,14 @@ def track_subscribed_build_completed
...
+      commit_status.set_failure(build.violations_count)
     else
-      commit_status.set_success(build.violation_count)
+      commit_status.set_success(build.violations_count)
     end
   end
 
   def fail_build?
-    hound_config.fail_on_violations? && build.violation_count > 0
+    hound_config.fail_on_violations? && build.violations_count > 0
```
Comment (gylaz) [inline app/services/build_report.rb]: Me thinks this is a bad rule for us to follow. ✂️ it?

Comment (thorncp) [inline app/services/build_report.rb]: I'd be okay removing it :+1:

**Comment by thorncp** [comment]
LGTM :traffic_light:

**Pull request merged**